src/main/java/com/olgaboiar/mint/Server.java

```diff
@@ -14,26 +19,30 @@
     String host;
     static int port;
     RequestParser parser = new RequestParser();
-    ResponseGenerator responseGenerator = new ResponseGenerator();
     Router router = new Router();
     Response response;
+    ILogger logger;
+    static String date = DateTimeFormatter.RFC_1123_DATE_TIME.format(ZonedDateTime.now());
+
 
-    public Server(String host, int port) {
+    public Server(String host, int port, ILogger logger) {
         this.host = host;
         this.port = port;
+        this.logger = logger;
     }
 
     @Override
     public void start() throws IOException {
         serverSocket = new ServerSocket(port);
+        logger.logMessage(date + "\nServer started.");
+        logger.logMessage("\nConnection on port " + port);
     }
 
     @Override
     public void run() throws IOException {
         acceptClientConnection();
         listenToClientConnection();
         readClientInput();
-//        Response response = responseGenerator.generateResponse();
         sendResponseToClient(response);
         closeClientConnection();
     }
```

Comment:
So I had a couple of thought about these method calls in in the `run`.
The first is that I'm not sure that they should be public methods. They're dependent steps in the request/response cycle, meaning a method like `listenToClientConnection` doesn't really stand up on it's own, which is a decent indicator that it might be better scoped as private

The other part I was thinking about was following the methods' interactions. By name it's obvious that these methods are interacting, but the implementation makes it hard to comprehend how. When I dig into the body of the methods I can see that they are leveraging interactions between different instance fields, but IMO it might be more clear if the instance variables were removed and  the void methods explicitly returned values to pass around amongst themselves.

Reply:
I'm also happy to follow up with you to talk more about this, it's hard to convey those thoughts well through a PR comment, but I've got some ideas for possible refactorings and it might clear any uncertainty I left in the previous comment.

Reply:
I'll try to refactor it in next PR

Reply:
Cool, let me know If you run into any hangups or have any questions.